src/lib.rs

```diff
@@ -275,6 +298,12 @@ mod test {
     }
 
     #[test]
+    fn parse_v4_non_zero_host_bits() {
+        let cidr = Ipv4Network::from_cidr("10.1.1.1/24");
+        assert!(cidr.is_err());
```

Comment:
I think I missed the point you tried to make with that link. I can't find a CIDR with non-zero host bits in that link. However I found that Apache Commons accepts it so I agree with your anyway. https://commons.apache.org/proper/commons-net/apidocs/org/apache/commons/net/util/SubnetUtils.html